erpnext/stock/doctype/stock_ledger_entry/stock_ledger_entry.py

```diff
@@ -172,6 +172,40 @@ def validate_with_last_transaction_posting_time(self):
 					msg += "<br>" + "<br>".join(authorized_users)
 					frappe.throw(msg, BackDatedStockTransaction, title=_("Backdated Stock Entry"))
 
+	def validate_future_sle_by_batch_for_outgoing_entries(self):
+		allow_neg = frappe.db.get_single_value('Stock Settings', 'allow_negative_stock')
+		if not self.batch_no or allow_neg or self.actual_qty is None or self.actual_qty >= 0:
+			return
```

Comment:
```suggestion
		if not self.batch_no or self.actual_qty is None or self.actual_qty >= 0:
			return
		if frappe.db.get_single_value('Stock Settings', 'allow_negative_stock'):
			return
```

Reducing unnecessary query.